data_structures/skip_list.cpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * A skip list is a data structure that is used for storing a sorted list of items with a
+ *  help of hierarchy of linked lists that connect increasingly sparse subsequences of the items
+ * 
+ *  References used: GeeksForGeeks skip list code, https://iq.opengenus.org/skip-list/ Pseudo Code.
+*/
+
+#include <iostream>
+#include <vector>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <time.h>   
+#include <cstdlib>
+
+// using namespace std; 
+using std::vector;
+using std::endl;
+
+#define MAXLVL 2
+#define P 0.5
+  
+
+/**
+ *  Node structure [Key][Node*, Node*...]
+*/
+class Node { 
+public: 
+    int key; 
+    void* value;
+    /*Forward Array*/
+    vector<Node*> forward;
+    Node(int key, int level, void* value); 
+};
+
+  
+Node::Node(int key, int level, void* value) { 
+    this->key = key; 
+  
+    /*Initialization of forward vector*/
+    for (int i = 0; i < sizeof(Node*)*(level+1); i++){
```

Comment:
from this definition, the loop termination condition is:
> till `i` is less than the platform's pointer size times ( the number of levels + 1). 

Does this even work? on modern machines, the `sizeof(Node*)` evaluates to 4. So, the loop will proceed 12 times when `level = MAXLVL`. Is this the intent here?

Reply:
Yeah, it should've been just for (int i = 0; i < (level+1); i++). You are correct.
Thank you again